test/parallel/test-quic-server-http3.js

```diff
@@ -0,0 +1,102 @@
+// Flags: --no-warnings
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+
+if (!common.hasQuic)
+  common.skip('quic support is not enabled');
+
+const {
+  Endpoint,
+} = require('net/quic');
+
+const {
+  TextDecoderStream,
+} = require('stream/web');
+
+const fixtures = require('../common/fixtures');
+
+const endpoint = new Endpoint({ address: { port: 0 } });
+
+endpoint.onsession = common.mustCall(({ session }) => {
+  session.onstream = common.mustCall(async ({ stream, respondWith }) => {
+    respondWith({
+      headers: { 'content-type': 'text/plain' },
+      trailers: function() {
+        return new Promise((res) => {
+          setTimeout(() => res({ abc: 2 }), 100);
+        });
```

Comment:
Shouldn't this use `setTimeout` from `timers/promises`?

Reply:
It can yes.